tests/test_repository_lib.py

```diff
@@ -526,6 +704,120 @@ def test_generate_timestamp_metadata(self):
+
+    # Set valid generate_timestamp_metadata() arguments.
+    version = 1
+    expiration_date = '1985-10-21T13:20:00Z'
+
+    # Load a valid repository so that top-level roles exist in roledb and
+    # generate_snapshot_metadata() has roles to specify in snapshot metadata.
+    repository = repo_tool.Repository(repository_directory, metadata_directory,
+        targets_directory, repository_name)
+
+    repository_junk = repo_tool.load_repository(repository_directory,
+        repository_name)
+
+    timestamp_metadata = repo_lib.generate_timestamp_metadata(snapshot_filename,
+        version, expiration_date, repository_name, use_length=False)
+    self.assertTrue(tuf.formats.TIMESTAMP_SCHEMA.matches(timestamp_metadata))
+
+    # Check that length is not calculated but hashes is
+    timestamp_file_info = timestamp_metadata['meta']
+
+    self.assertTrue(timestamp_file_info['snapshot.json'].get('length') is None)
+    self.assertTrue(timestamp_file_info['snapshot.json'].get('hashes') is not None)
+
+
+
+  def test_generate_timestamp_metadata_without_hashes(self):
+    # Test normal case.
+    repository_name = 'test_repository'
+    temporary_directory = tempfile.mkdtemp(dir=self.temporary_directory)
+    original_repository_path = os.path.join('repository_data',
+                                            'repository')
+    repository_directory = os.path.join(temporary_directory, 'repository')
+    shutil.copytree(original_repository_path, repository_directory)
+    metadata_directory = os.path.join(repository_directory,
+                                      repo_lib.METADATA_STAGED_DIRECTORY_NAME)
+    targets_directory = os.path.join(repository_directory, repo_lib.TARGETS_DIRECTORY_NAME)
+
+    snapshot_filename = os.path.join(metadata_directory,
+                                     repo_lib.SNAPSHOT_FILENAME)
+
+    # Set valid generate_timestamp_metadata() arguments.
+    version = 1
+    expiration_date = '1985-10-21T13:20:00Z'
+
+    # Load a valid repository so that top-level roles exist in roledb and
+    # generate_snapshot_metadata() has roles to specify in snapshot metadata.
+    repository = repo_tool.Repository(repository_directory, metadata_directory,
+        targets_directory, repository_name)
+
+    repository_junk = repo_tool.load_repository(repository_directory,
+        repository_name)
+
+    timestamp_metadata = repo_lib.generate_timestamp_metadata(snapshot_filename,
+        version, expiration_date, repository_name, use_hashes=False)
+    self.assertTrue(tuf.formats.TIMESTAMP_SCHEMA.matches(timestamp_metadata))
+
+    # Check that hashes is not calculated but length is
+    timestamp_file_info = timestamp_metadata['meta']
+
+    self.assertTrue(timestamp_file_info['snapshot.json'].get('length') is not None)
+    self.assertTrue(timestamp_file_info['snapshot.json'].get('hashes') is None)
+
+
+
+  def test_generate_timestamp_metadata_without_length_and_hashes(self):
```

Comment:
Ditto: would it be possible to reuse the logic from the individual tests above?

Reply:
I rebased and addressed that comment.